onnxruntime/core/providers/openvino/ov_factory.cc

```diff
@@ -0,0 +1,182 @@
+
+  // Create a map for device type mapping
+  static const std::map<OrtHardwareDeviceType, std::string> ort_to_ov_device_name = {
+      {OrtHardwareDeviceType::OrtHardwareDeviceType_CPU, "CPU"},
+      {OrtHardwareDeviceType::OrtHardwareDeviceType_GPU, "GPU"},
+      {OrtHardwareDeviceType::OrtHardwareDeviceType_NPU, "NPU"},
+  };
+
+  for (size_t i = 0; i < num_devices && num_ep_devices < max_ep_devices; ++i) {
+    const OrtHardwareDevice& device = *devices[i];
+    if (ort_api.HardwareDevice_VendorId(&device) != vendor_id_) {
+      // Not an Intel Device.
+      continue;
+    }
+
+    auto device_type = ort_api.HardwareDevice_Type(&device);
+    auto device_it = ort_to_ov_device_name.find(device_type);
+    if (device_it == ort_to_ov_device_name.end()) {
+      // We don't know about this device type
+      continue;
+    }
+
+    const auto& ov_device_type = device_it->second;
+    std::string ov_device_name;
+    auto get_pci_device_id = [&](const std::string& ov_device) {
+      try {
+        ov::device::PCIInfo pci_info = ov_core_->get_property(ov_device, ov::device::pci_info);
+        return pci_info.device;
+      } catch (ov::Exception&) {
+        return 0u;  // If we can't get the PCI info, we won't have a device ID.
+      }
+    };
+
+    auto filtered_devices = GetOvDevices(ov_device_type);
+    auto matched_device = filtered_devices.begin();
+    if (filtered_devices.size() > 1 && device_type == OrtHardwareDeviceType::OrtHardwareDeviceType_GPU) {
+      // If there are multiple devices of the same type, we need to match by device ID.
+      matched_device = std::find_if(filtered_devices.begin(), filtered_devices.end(), [&](const std::string& ov_device) {
+        uint32_t ort_device_id = ort_api.HardwareDevice_DeviceId(&device);
+        return ort_device_id == get_pci_device_id(ov_device);
+      });
+    }
+
+    if (matched_device == filtered_devices.end()) {
+      // We didn't find a matching OpenVINO device for the OrtHardwareDevice.
+      continue;
+    }
+
+    // these can be returned as nullptr if you have nothing to add.
+    OrtKeyValuePairs* ep_metadata = nullptr;
+    OrtKeyValuePairs* ep_options = nullptr;
+    ort_api.CreateKeyValuePairs(&ep_metadata);
+    ort_api.AddKeyValuePair(ep_metadata, ov_device_key_, matched_device->c_str());
+
+    if (IsMetaDeviceFactory()) {
+      ort_api.AddKeyValuePair(ep_metadata, ov_meta_device_key_, device_type_.c_str());
+    }
+
+    // Create EP device
+    auto* status = ort_api.GetEpApi()->CreateEpDevice(this, &device, ep_metadata, ep_options,
+                                                      &ep_devices[num_ep_devices++]);
+
+    ort_api.ReleaseKeyValuePairs(ep_metadata);
+    ort_api.ReleaseKeyValuePairs(ep_options);
+
+    if (status != nullptr) {
+      return status;
+    }
+  }
+
+  return nullptr;
+}
+
+extern "C" {
+//
+// Public symbols
+//
+OrtStatus* CreateEpFactories(const char* /*registration_name*/, const OrtApiBase* ort_api_base,
```

Comment:
needs update because of https://github.com/microsoft/onnxruntime/pull/25456

Reply:
fixed in latest commit.